fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/StatisticsCalculator.java

```diff
@@ -680,7 +733,16 @@ private Void computeProjectNode(ExpressionContext context, Map<ColumnRefOperator
 
         for (ColumnRefOperator requiredColumnRefOperator : columnRefMap.keySet()) {
             ScalarOperator mapOperator = columnRefMap.get(requiredColumnRefOperator);
-            ColumnStatistic outputStatistic = ExpressionStatisticCalculator.calculate(mapOperator, allBuilder.build());
+            if (mapOperator instanceof SubfieldOperator && context.getOptExpression() != null) {
+                List<Operator> scanOperators = Utils.collectScanOperators(context.getOptExpression());
```

Comment:
poor performance.
and the statistic is error, doesn't compute child node and predicate

Reply:
Yes，I known, but we don't support prune subfield of strcut at logical rewrite stage, can not cpmute stats of subfield at scan node